weavenet/readme.adoc

```diff
@@ -0,0 +1,315 @@
+
+  networking:
+    weave: {}
+
+Rest of the chapter goes with the presumption that you are using an existing cluster.
+
+NOTE: A smaller sized cluster, with 1 master node and 2 worker nodes, is chosen here. This is to minimize the amount of time for rolling update the cluster later. It typically takes 5-6 mins per node for rolling update.
+
+All configuration files for this chapter are in the `weave` directory. Make sure you change to that directory before giving any commands in this chapter.
+
+== Installation
+
+Edit the Kubernetes cluster configuration using the command:
+
+  kops edit cluster example.cluster.k8s.local
+
+By default, Kops uses `kubenet` for networking. This is captured in the configuration file as:
+
+  networking:
+    kubenet: {}
+
+Update the networking configuration to use `weave` by setting the following property:
+
+  networking:
+    weave: {}
+
+View changes that will be applied to the cluster:
+
+  kops update cluster example.cluster.k8s.local
+
+It shows output as:
+
+```
+I1025 15:55:45.512183    3454 apply_cluster.go:420] Gossip DNS: skipping DNS validation
+[...]
+
+Will modify resources:
+  LaunchConfiguration/master-us-east-1d.masters.cluster.k8s.local
+    UserData
+                          ...
+                            - _aws
+                            - _kubernetes_master
+                          + - _networking_cni
+                            channels:
+                            - s3://kubernetes-aws-io/cluster.k8s.local/addons/bootstrap-channel.yaml
+                          ...
+
+
+  LaunchConfiguration/nodes.cluster.k8s.local
+    UserData
+                          ...
+                            - _automatic_upgrades
+                            - _aws
+                          + - _networking_cni
+                            channels:
+                            - s3://kubernetes-aws-io/cluster.k8s.local/addons/bootstrap-channel.yaml
+                          ...
+
+
+  LoadBalancer/api.cluster.k8s.local
+    Lifecycle              <nil> -> Sync
+
+  LoadBalancerAttachment/api-master-us-east-1d
+    Lifecycle              <nil> -> Sync
+
+
+Must specify --yes to apply changes
+```
+
+Apply the changes using the command:
+
+  kops update cluster example.cluster.k8s.local --yes
+
+It shows the output:
+
+```
+I1025 15:56:26.679683    3458 apply_cluster.go:420] Gossip DNS: skipping DNS validation
+[...]
+Kops has set your kubectl context to example.cluster.k8s.local
+
+Cluster changes have been applied to the cloud.
+
+
+Changes may require instances to restart: kops rolling-update cluster
+```
+
+Determine if any of the nodes will require a restart using the command:
+
+  kops rolling-update cluster example.cluster.k8s.local
+
+Output from this command is shown:
+
+```
+$ kops rolling-update cluster example.cluster.k8s.local
+NAME              STATUS      NEEDUPDATE  READY MIN MAX NODES
+master-us-east-1d NeedsUpdate 1           0     1   1   1
```

Comment:
Existing cluster configuration is 3 master and 5 worker nodes. Can you update the output accordingly?

Reply:
This is in the context the user has been [instructed to create a smaller cluster](https://github.com/aws-samples/aws-workshop-for-kubernetes/pull/255/files#diff-0c64f05689728a097473a0fadab32bb2R37) and is now updating it.  That text was copied verbatim from the calico variant.

However #242 points out this update operation is unsupported by `kops`, so I'm re-doing the Weave Net doc to take it out.